lib/aee.js

```diff
@@ -0,0 +1,69 @@
+'use strict';
+
+const { iter } = require('@metarhia/common');
+
+class AsyncEventEmitter {
+  constructor() {
+    this.events = new Map();
+    this.wrappers = new Map();
+  }
+
+  on(name, fn) {
+    let event = this.events.get(name);
+    if (!event) {
+      event = new Set();
+      this.events.set(name, event);
+    }
+    event.add(fn);
+  }
+
+  once(name, fn) {
+    const wrapper = (...args) => {
+      this.remove(name, wrapper);
+      return fn(...args);
+    };
+    this.wrappers.set(fn, wrapper);
+    this.on(name, wrapper);
+  }
+
+  async emit(name, ...args) {
+    const event = this.events.get(name);
+    if (!event) return undefined;
+    const listeners = event.values();
+    const promises = iter(listeners).map(fn => fn(...args));
+    return Promise.all(promises);
+  }
+
+  remove(name, fn) {
+    const event = this.events.get(name);
+    if (!event) return;
+    if (event.has(fn)) {
+      event.delete(fn);
+    } else {
+      const wrapper = this.wrappers.get(fn);
```

Comment:
IMHO, you should also clean `this.wrappers` because otherwise, it will only grow.